float32 dist
float32 theta1
float32 theta2
